src/components/v5/common/Table/Table.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState } from 'react';
+import {
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+  createColumnHelper,
+} from '@tanstack/react-table';
+import { useFieldArray, useFormContext } from 'react-hook-form';
+import clsx from 'clsx';
+import { v4 as uuidv4 } from 'uuid';
+import { TableProps } from './types';
+import Button from '~v5/shared/Button';
+import { useMobile } from '~hooks';
+import BurgerMenu from '~v5/common/Table/partials/BurgerMenu';
+
+const displayName = 'v5.common.Table';
+
+const Table = <T,>({
+  className,
+  tableTitle,
+  columns,
+  action,
+  isMenuVisible,
+  onToogle,
+  onToogleOff,
+  registerContainerRef,
+}: TableProps<T>) => {
+  const isMobile = useMobile();
+  const { type, actionData, actionText } = action;
+  const [selectedRowId, setSelectedRowId] = useState<string>();
+  const { control, getValues } = useFormContext();
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: type,
+  });
```

Comment:
FieldArray should be outside of this component. This table is meant to be dummy component not a field